src/main.rs

```diff
@@ -135,6 +137,68 @@ fn work() -> Result<(), Cue2CCDError> {
         _ => return Err(Cue2CCDError::InvalidProtectionError {}),
     };
 
+    // Technically speaking, there's no reason you *shouldn't* be able to provide an SBI/LSD
+    // file even if you didn't choose protection
+
+    // SBI File Format:
+    // Starts with header 0x53 0x42 0x49 0x00 ('S' 'B' 'I' '0x00')
+    // The entire rest of the file consists of subQ data, specifically consisting of the actual
+    // MSF current subQ was read from, followed by a dummy 0x01 byte, followed by the first 10 bytes
+    // of that subQ (so, everything but the CRC16) The exclusion of the CRC16 is obviously
+    // annoying, *especially*  for SecuROM and LibCrypt. LSD is a better file format, but at the
+    // moment, redump will only  generate LSD files for PS1 discs, and we do not have the power to
+    // change the website; so, until a successor website exists, SBI support is necessary. It's
+    // also still preferred by a lot of people and emulators for PS1 for some reason, despite
+    // being worse than LSD.
+
+    //let mut sbi_data =  Some(vec<u8>);
+    // Some(output_stem.with_extension("sbi").file_name());
+
+    let mut sbi_lba_array: Vec<i32> = Vec::new();
+    let mut sbi_data: Vec<Vec<u8>> = Vec::new();
+    // SBI files have never been defined in the cuesheet, and programs (mainly just PS1
+    // emulators so far) that make use of them simply check if there's an SBI file with the
+    // same basename next to the .cue. If one exists, they use it, otherwise they don't.
+    // It seems  best to keep in line with this behavior
+
+    // TODO: is this extension check case sensitive?
+    if Path::new(&output_stem.with_extension("sbi")).exists() {
+        // SBI files are very small, so it seems best to read the whole thing in first?
+        let raw_sbi_data = Some(std::fs::read(Path::new(
+            &output_stem.with_extension("sbi"),
+        ))?)
+        .unwrap(); // Already confirmed it was something, so, this should be fine?
+        let (header, data) = raw_sbi_data.split_at(4);
+
+        if header != [83, 66, 73, 00] {
+            // Checks for required "[S][B][I][0x00] header
+            println!("not equal"); // Not sure what to do if for some reason not there
+        }
+        // should always be multiple of 14
+        for (chunkIndex, chunk) in data.chunks(14).enumerate() {
```

Comment:
Looks like `chunkIndex` didn't get used, so you can probably just iterate over `data.chunks()`?
```suggestion
        for chunk in data.chunks(14) {
```

Reply:
Yeah, sorry. I think i had that in when I was trying to see if there was any way i could get something like a Vec that was FIFO, but it seems like all the options for that are more trouble than they're worth, at least as opposed to just iterating through once this all gets passed to lib.rs.